registry/consul/service_discovery.go

```diff
@@ -141,9 +137,10 @@ func (csd *consulServiceDiscovery) Register(instance registry.ServiceInstance) e
 		consulClient *consul.Client
 	)
 	ins, _ := csd.buildRegisterInstance(instance)
-	if consulClient, err = csd.getConsulClient(); err == nil {
-		err = consulClient.Agent().ServiceRegister(ins)
+	if consulClient = csd.getConsulClient(); consulClient == nil {
+		return perrors.New("consul client is closed!")
```

Comment:
var (
    ConsulClientClosed = perrors.New("consul client is closed")
)

Reply:
lgtm